src/DataDryad.jl

```diff
@@ -0,0 +1,42 @@
+immutable DataDryad <: DataRepo
+end
+
+base_url(::DataDryad) = "https://datadryad.org/mn/"
+
+function description(::DataDryad, mainpage)
+    desc = replace(text_only(first(matchall(sel".article-abstract", mainpage.root))), "Abstract ", "")
+    author = text_only(first(matchall(sel".pub-authors a", mainpage.root)))
+    license = getattr(first(matchall(sel".single-image-link", mainpage.root)), "href")
+    date = now()
+    paper = text_only(first(matchall(sel".citation-sample a", mainpage.root)))
+    dataset = replace(text_only(last(matchall(sel".publication-header p", mainpage.root))), "DOI: ", "")
+
+    final = escape_multiline_string("""
+    Author: $(author) et. al.
+    License: $(license)
+    Date: $(date)
+
+    $(desc)
+
+    Please cite this paper:
+    $(paper)
+    as well as this dataset:
+    $(dataset)
+    if you use this in your research.
+    """, "\$")
+end
+
+function get_urls(repo::DataDryad, page)
+    urls = []
+    paper = text_only(last(matchall(sel".publication-header p", page.root)))
+    download_url = replace(paper, "DOI: https://", "https://datadryad.org/mn/object/http://dx.") * "/1/bitstream"
+    push!(urls, download_url)
+    md5 = ("md5", text_only(getpage(replace(paper, "DOI: https://doi.org/", "https://datadryad.org/mn/checksum/doi:") * "/1").root))
```

Comment:
This isn't going to workout right:
the download urls and the md5 s are a different argument.
This produces: (in the example below)

```
Any["https://datadryad.org/mn/object/http://dx.doi.org/10.5061/dryad.74699/1/bitstream", ("md5", "bc96dba38d8659f42527cc22ff7f6e3b")]
```

But what we want is:

```
Any["https://datadryad.org/mn/object/http://dx.doi.org/10.5061/dryad.74699/1/bitstream"],
Any[("md5", "bc96dba38d8659f42527cc22ff7f6e3b")]
```

Fixing this is going to mean doing a bit more reworking, than is needed in this PR.
we should separate `get_checksums` from `get_urls`,
and then modify the main method around
https://github.com/oxinabox/DataDepsGenerators.jl/blob/master/src/DataDepsGenerators.jl#L31
to call it.
But we'll also need to provide correct handling for the other generators, which do not (currently),
define  `get_checksums`.

Further to that, I'm smelling an a bit of a code-smell in my code in that area.
So leave it for now, by just commenting out the md5 stuff,
and I'll propose a refactoring PR after this is done.

Reply:
Yes, I was skeptical of changing the structure of DataDepsGenerators, which I would have to do, if I was implementing in the format you suggested. That's why, I did it this way.